daras_ai_v2/base.py

```diff
@@ -490,39 +574,49 @@ def _render_published_run_save_buttons(self, *, sr: SavedRun, pr: PublishedRun):
             )
 
             if can_edit:
-                ref = gui.use_alert_dialog(key="options-modal")
-                if gui.button(
-                    label=icons.fork_lg, className="mb-0 ms-lg-2", type="tertiary"
-                ):
-                    ref.set_open(True)
-                if ref.is_open:
-                    with gui.alert_dialog(ref=ref, modal_title="#### Options"):
-                        self._saved_options_modal(sr=sr, pr=pr)
-                label = "Update"
+                icon, label = icons.save, "Update"
+            elif self._has_request_changed():
+                icon, label = icons.save, "Save and Run"
             else:
-                label = "Save"
+                icon, label = icons.fork, "Save as New"
 
-            ref = gui.use_confirm_dialog(key="publish-modal", close_on_confirm=False)
+            ref = gui.use_alert_dialog(key="publish-modal")
             if gui.button(
-                f'<i class="fa-regular fa-floppy-disk"></i> <span class="d-none d-lg-inline">{label}</span>',
+                f'{icon} <span class="d-none d-lg-inline">{label}</span>',
                 className="mb-0 ms-lg-2 px-lg-4",
                 type="primary",
             ):
+                if not self.request.user or self.request.user.is_anonymous:
```

Comment:
please put this in an if..else case |  its not clear from this that publish_for_anonymous_user will end up raising an exception and the stuff below won't run

Reply:
i added a type annotation for `_publish_for_anonymous_user() -> NoReturn` for this, but i like this because it's even more explicit